InCanvasSearch: fix context menu

aosyatnik opened the pull request:
### Purpose

I found bug. InCanvasSearch, which is part of context menu, doesn't create node at the same place, where user clicked. That is because InCanvasSearch can't TranslatePoint properly, when its' parent is Context menu.

So, I have to add `InCanvasSearchPosition`, which stores position of mouse, when user clicked on canvas.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] The level of testing this PR includes is appropriate

### Reviewers

@Benglin 
@pboyer 
@ramramps

ramramps commented:
@aosyatnik - InCanvasSearchPosition property has to be in Viewmodel and not in the view. 
So instead of getting the  InCanvasSearchControl object in workspaceview.xaml.cs, you can set the property on InCanvasSearch Viewmodel. May be you can have a propery in WorkspaceView.xaml.cs that returns InCanvasSearch Viewmodel. I think you can test this behavior in that case. 

What say?

aosyatnik commented:
Hi, @ramramps . I thought about it. So, what I've done:
1. Created in `WorkspaceViewModel` 2 ViewModels : `InCanvasSearchContextMenuViewModel`  and `InCanvasSearchBarViewModel` . First is used in context menu, second is used in search bar, that can be called by Shift+DoubleClick.
2. Added to `SearchViewModel` new property `InCanvasSearchPosition`, that indicates point, where canvas was clicked.
3. Removed search view model stuff from `WorkspaceView`.

Please, take a look. Thanks.

ramramps commented:
@aosyatnik Good. LGTM !
